src/shared/locked-status/use-check-lock-status.js

```diff
@@ -10,96 +12,138 @@ import { useDataValueSet } from '../use-data-value-set/use-data-value-set.js'
     const dataValueSet = useDataValueSet()
+    const { setLockStatus } = useLockedContext()
 
     useEffect(() => {
-        if (
-            isDataInputPeriodLocked({
-                dataSetId,
-                periodId,
-                metadata,
-                currentDayString,
-            })
-        ) {
-            // mark as invalid for data input period
-            setLockStatus(LockedStates.LOCKED_DATA_INPUT_PERIOD)
+        // prefer lock status from backend, found in dataValueSet
+        const backendLockStatus =
+            BackendLockStatusMap[dataValueSet.data?.lockStatus]
+        if (backendLockStatus) {
+            setLockStatus({ state: backendLockStatus })
             return
         }
 
-        // else default to lockStatus from dataValueSet
-        if (BackendLockStatusMap[dataValueSet.data?.lockStatus]) {
-            setLockStatus(BackendLockStatusMap[dataValueSet.data?.lockStatus])
+        // if either 1. backend status is 'OPEN' or 2. it's not defined yet,
+        // refine the lock status here from properties on the dataSet:
+        // (a lock status of 'OPEN' from the backend could mean either that the
+        // form is open OR that the form should be locked due to data input
+        // period.)
+        // Therefore, check the dataInputPeriod boundaries, and if the form IS
+        // open, get the date the form will close, if applicable.
+        const frontendLockStatus = getFrontendLockStatus({
+            dataSetId,
+            selectedPeriod,
+            metadata,
+            clientServerDateUtils,
+        })
+        if (frontendLockStatus) {
+            setLockStatus(frontendLockStatus)
             return
         }
 
         // otherwise denote as open
-        setLockStatus(LockedStates.OPEN)
+        setLockStatus({ state: LockedStates.OPEN })
     }, [
         metadata,
         dataSetId,
         orgUnitId,
-        periodId,
+        clientServerDateUtils,
         dataValueSet.data?.lockStatus,
         setLockStatus,
-        currentDayString,
+        selectedPeriod,
     ])
 }
-
-export const updateLockStatusFromBackend = (
-    frontEndLockStatus,
-    backEndLockStatus,
-    setLockStatus
-) => {
-    // if the lock status is APPROVED, set to approved
-    if (backEndLockStatus === 'APPROVED') {
-        setLockStatus(LockedStates.LOCKED_APPROVED)
-        return
-    }
-
-    // if the lock status is LOCKED, this is locked due to expiry days
-    if (backEndLockStatus === 'LOCKED') {
-        setLockStatus(LockedStates.LOCKED_EXPIRY_DAYS)
-        return
-    }
-
-    // a lock status of 'OPEN' from the backend could mean either that the form is open OR
-    // that the form should be locked due to data input period, SO
-    // set to OPEN unless frontend check has identified that data input period as out-of-bounds
-    if (frontEndLockStatus !== LockedStates.LOCKED_DATA_INPUT_PERIOD) {
-        setLockStatus(LockedStates.OPEN)
-    }
-}
```

Comment:
This was mostly integrated in `useCheckLockStatus` due to the `useEffect` dependency on `dataValueSet.data?.lockStatus`, so tried to let that handle the backend check